File: smartlist/src/main/java/com/smartlist/module/auth/AuthRepository.java
package com.smartlist.module.auth;

import com.smartlist.model.AuthorizedToken;
import com.smartlist.model.Candidate;
import com.smartlist.model.User;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface AuthRepository extends CrudRepository<AuthorizedToken, Integer> {

    Optional<AuthorizedToken> findByAccessTokenHash(String accessTokenHash);

    void deleteByUser(User user);

    void deleteByCandidate(Candidate candidate);
}

File: javascript/smartlist/smartlist-api-mysql-project/src/modules/auth/auth.repository.ts
import { EntityRepository, Repository } from 'typeorm';
import { Injectable } from '@nestjs/common';
import { AuthorizedToken } from 'src/entities/authorized-token.entity';
import { User } from 'src/entities/user.entity';
import { Candidate } from 'src/entities/candidate.entity';

@Injectable()
@EntityRepository(AuthorizedToken)
export class AuthRepository extends Repository<AuthorizedToken> {
  async findByAccessToken(accessTokenHash: string): Promise<AuthorizedToken> {
    return await this.findOne({
      where: { access_token_hash: accessTokenHash },
    });
  }

  async deleteByUser(user: User) {
    await this.delete({ user: user });
  }

  async deleteByCandidate(candidate: Candidate) {
    await this.delete({ candidate: candidate });
  }
}
